actions/traits/LtiModuleTrait.php

```diff
@@ -52,16 +53,28 @@ protected function returnError($description, $returnLink = true) {
                     $this->setData('consumerLabel', $consumerLabel);
                 }
 
-                if($launchData->hasVariable(\taoLti_models_classes_LtiLaunchData::LAUNCH_PRESENTATION_RETURN_URL)) {
-                    $this->setData('returnUrl', $launchData->getReturnUrl());
+                if ($launchData->hasVariable(\taoLti_models_classes_LtiLaunchData::LAUNCH_PRESENTATION_RETURN_URL)) {
+                    $flowController = new FlowController();
+                    $flowController->redirect($this->getLtiReturnUrl($launchData, $error));
                 }
-            } catch (\taoLti_models_classes_LtiException $exception) {
-                // no Lti Session started
             }
+
             if (!empty($description)) {
```

Comment:
this variable isn't defined, so end user has no error if we don't set return url